dali/operators/generic/reduce/mean_square.h

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_GENERIC_REDUCE_MEAN_SQUARE_H__
+#define DALI_OPERATORS_GENERIC_REDUCE_MEAN_SQUARE_H__
+
+#include "include/dali/core/static_map.h"
+#include "dali/operators/generic/reduce/reduce.h"
+
+/// Definition of mapping between input types and possible output types
+/// for MeanSquare operator. Is uesed as parameter to TYPE_MAP macro.
+#define MEAN_SQUARE_TYPES_MAP ( \
+    ((uint8_t), (uint8_t, float)), \
+    ((int8_t), (int8_t, float)), \
+    ((uint16_t), (uint16_t, float)), \
+    ((int16_t), (int16_t, float)), \
+    ((uint32_t), (uint32_t, float)), \
+    ((int32_t), (int32_t, float)), \
+    ((uint64_t), (uint64_t, float)), \
+    ((int64_t), (int64_t, float)), \
```

Comment:
Calculating mean square and staying in a small output type is asking for trouble - by squaring we're doubling the number of bits required. We also _can_ support both signed and unsigned outputs for signed input just by reinterpreting the data - that would complicate the matters a bit, but would cut the number of instantiations without sacrificing functionality (compared to either supporting both signed and unsigned outputs or not supporting one of the type flavors).